### ISSUE TITLE
This has become unusable because of the endless and repeated prompts to login to azure even when not using the tool

### ISSUE THREAD

**Issue by JohnGalt1717** [vscode-kubernetes-tools/vscode-kubernetes-tools#1217]
The tool produces a prompt per registration of a Kubernetes cluster. This results in me having 6 separate prompts TO THE EXACT SAME LOGIN.

Worse, it prompts when you're not even using the Kubernetes plugin.  And if you don't respond to the endless prompts, it does it over and over again periodically.

Please stop the madness and fix this so that the tool can actually be used:

1. 1 prompt per unique tenant if Azure. (same for any others)
2. Don't prompt if not using the tool.
3. Don't reprompt if the person doesn't respond.
4. Fix the lock out if you try and do anything and the login hasn't worked. It should fail gracefully.

**Comment by peterbom** [comment]
Hi @JohnGalt1717, and thanks for the feedback!

To me, this initially sounded similar to #1191, but I'd like to check with you to be sure we're talking about the same thing. How do the prompts appear? Are they within the VS Code UI, or is VS Code attempting to launch a browser window to perform Azure authentication for an AAD-enabled cluster (several times for the same cluster)?

There are a couple of things in your description which make me wonder if this is something different, though (the above issue prompts several times _per cluster_, it only occurs when using the Kubernetes plugin, and I don't think it keeps nagging you if you're not actually interacting with the cluster treeview).

The following would be really helpful to help us narrow down the issue:
- Would it be possible for you to share some screenshots or copy the text you're seeing in the prompts?
- Does it specifically happen for AAD-enabled clusters?
- Did you use the AKS extension to add these clusters to your `kubeconfig` file?
- What OS are you using and what's the extension version?

**Comment by jheran007** [comment]
@JohnGalt1717 did you ever find a resolution to this? I'm having the same experience.

**Comment by JohnGalt1717** [comment]
Nope. Still spews a login prompt per k8s cluster registered despite the fact that they're all Azure under the same account.

**Comment by jheran007** [comment]
As a test, I deleted my ./kube/config file and then merged the clusters back into the my kubeconfig via the extension. So far, this has alleviated the shotgun blast of login prompts.

**Comment by specialforest** [comment]
Just had the same issue. After inspecting kube config and removing an orphan cluster from it the interactive login popped one more time and VS Code doesn't ask more after restart 🥳
